admin_interface/models.py

```diff
@@ -8,18 +8,41 @@
 from .cache import del_cached_active_theme
 
 
+class ThemeQuerySet(models.QuerySet):
+    def get_active_theme(self):
+        objs_active_qs = self.filter(active=True)
+        objs_active_ls = list(objs_active_qs)
+        objs_active_count = len(objs_active_ls)
```

Comment:
Hm, loading all themes into a list could be avoided, for example by using `active_objects.count()` then `first()` or `last()` method.  Maybe it does not matter much given the very low volume of data though (and caching).

Also, the algo could be optimized for the most common case where there is 1 active theme (meaning change the order of the tests below).

Reply:
This was a copy paste of the old code, but I agree that this is somehow overcomplicated.
We can use get_or_create and and add a Check Constraint which ensure that we can't have two active themes at the same time?

Reply:
@merwok you're right, but since the query should never return more than 1 or 2 objects, I think it's better to avoid the count query.